Spring-Boot/Spring-Boot-Exam/src/test/java/tw/waterball/judgegirl/springboot/exam/controllers/HomeworkControllerTest.java

```diff
@@ -0,0 +1,133 @@
+import tw.waterball.judgegirl.springboot.exam.SpringBootExamApplication;
+import tw.waterball.judgegirl.springboot.exam.view.HomeworkView;
+import tw.waterball.judgegirl.springboot.profiles.Profiles;
+import tw.waterball.judgegirl.submissionapi.clients.SubmissionServiceDriver;
+import tw.waterball.judgegirl.testkit.AbstractSpringBootTest;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ActiveProfiles(Profiles.JWT)
+@ContextConfiguration(classes = SpringBootExamApplication.class)
+public class HomeworkControllerTest extends AbstractSpringBootTest {
+
+    private static final String HOMEWORK_NAME = "homeworkName";
+    private static final String HOMEWORK_PATH = "/api/homework";
+
+    @Autowired
+    private FakeProblemServiceDriver problemServiceDriver;
+
+    @MockBean
+    private SubmissionServiceDriver submissionServiceDriver;
+
+    @AfterEach
+    void cleanUp() {
+        problemServiceDriver.clear();
+    }
+
+    @Test
+    public void GivenThreeProblemsCreated_WhenCreateHomeworkWithProblemIds_ShouldCreateSuccessfully() throws Exception {
+        Integer[] createProblemIds = {1, 2, 3};
+        createProblems(createProblemIds);
+
+        HomeworkView homework = getBody(createHomework(HOMEWORK_NAME, createProblemIds)
+                .andExpect(status().isOk()), HomeworkView.class);
+
+        List<String> homeworkProblemIds = homework.problemIds;
+        assertEquals(createProblemIds.length, homeworkProblemIds.size());
+        for (int i = 0; i < homeworkProblemIds.size(); i++) {
+            assertEquals(String.valueOf(createProblemIds[i]), homeworkProblemIds.get(i));
+        }
+    }
+
+    @DisplayName("Given created two problems [0, 1], " +
+            "When create homework with two problems id [0, 1] and a non existing problem id 2, " +
+            "Should respond two problems id in the homework")
+    @Test
+    public void testAddProblemsIntoHomeworkByNonExistingProblemIdAndTwoCreatedProblemIds() throws Exception {
+        Integer[] createProblemIds = {0, 1};
+        createProblems(createProblemIds);
+
+        int nonExistingProblemId = 2;
+        Integer[] testCaseProblemIds = {0, 1, nonExistingProblemId};
+        HomeworkView homework = getBody(createHomework(HOMEWORK_NAME, testCaseProblemIds)
+                .andExpect(status().isOk()), HomeworkView.class);
+
+        List<String> homeworkProblemIds = homework.problemIds;
+        assertEquals(createProblemIds.length, homeworkProblemIds.size());
+        for (int index = 0; index < homeworkProblemIds.size(); index++) {
+            assertEquals(String.valueOf(createProblemIds[index]), homeworkProblemIds.get(index));
+        }
```

Comment:
align your code with the test case's name as best as possible

so this test case can be read more easily

you should write something like:

```java
Integer[] problemIds = {0, 1, 2};

HomeworkView homework = createHomeworkAndGet(HOMEWORK_NAME, problemIds);

homeworkShouldIncludeProblemIds(homework, problemIds)
```